src/components/views/dialogs/spotlight/SpotlightDialog.tsx

```diff
@@ -1102,7 +1154,8 @@ const SpotlightDialog: React.FC<IProps> = ({ initialText = "", initialFilter = n
                     }
 
                     const idx = refs.indexOf(rovingContext.state.activeRef);
-                    ref = findSiblingElement(refs, idx + (accessibilityAction === KeyBindingAction.ArrowUp ? -1 : 1));
+                    const nextElementIndex = findNextElementIndex({ accessibilityAction, idx, refs });
+                    ref = findSiblingElement(refs, nextElementIndex);
```

Comment:
I'd rather see a function that calculates the offset to jump, and then see the clamping be done at this place, something like (pseudocode):

```suggestion
                    const nextElementOffset = calculateScrollOffset(accessibilityAction);
                    const nextElementIndex = Math.max(0, Math.min(refs.length - 1, idx + nextElementOffset));
                    ref = findSiblingElement(refs, nextElementIndex);
```